### PR TITLE
PLA-4041 list no ignore error

### PR DESCRIPTION
**Pull request by talltea**
# Citrine Python PR

## Description 
Removes the exception catching for elements in collection when listing. The exception catching was swallowing actual list errors causing e2es to create extra resources.
Adds the `module_type` to Workflow list endpoint. There are two (`DESIGN` and `PERFORMANCE`), so I chained the generators for those endpoints.

I haven't written tests yet, but I have run this locally against projects on development, and was able to get the correct workflows listing.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [ ] I have bumped the version in setup.py

### CODE REVIEW

File: src/citrine/_rest/collection.py [hunk 1]
```diff
@@ -199,13 +199,7 @@ def _build_collection_elements(self,
...
         """
         for element in collection:
-            try:
-                yield self.build(element)
-            except(KeyError, ValueError):
-                # TODO:  Right now this is a hack.  Clean this up soon.
-                # Module collections are not filtering on module type
-                # properly, so we are filtering client-side.
-                pass  # pragma: no cover
+            yield self.build(element)
```
Comment (maxhutch) [inline src/citrine/_rest/collection.py:202]: <3

File: src/citrine/resources/workflow.py [hunk 2]
```diff
@@ -28,9 +31,82 @@ def __init__(self, project_id: UUID, session: Session):
         self.project_id = project_id
         self.session: Session = session
 
+    def list(self,
+             page: Optional[int] = None,
+             per_page: int = 100) -> Iterable[ResourceType]:
```
Comment (maxhutch) [inline src/citrine/resources/workflow.py:36]:
```suggestion
             per_page: int = 100) -> Iterable[Workflow]:
```
We know the concrete type here, right?

**Review by maxhutch — COMMENTED** [review]
Looks right.  Does this work with the e2es?

**Comment by talltea** [comment]
> Looks right. Does this work with the e2es?

Yes, I just tested it by running our nextgen-ITs against development, and they passed: https://jenkins.corp.citrine.io/job/nextgen-ITs/11119/

**Review by maxhutch — DISMISSED** [review]
LGTM

**Review by maxhutch — APPROVED** [review]

**Pull request merged**